packages/core/src/external/aws/lambda-logic/bundle-to-html.ts

```diff
@@ -1577,10 +1579,12 @@ function createObservationLaboratorySection(observations: Observation[]) {
     const bDate = dayjs(b.effectiveDateTime).format(ISO_DATE);
     const aText = a.code?.text;
     const bText = b.code?.text;
-    if (aText === undefined || bText === undefined) {
+    const aCode = a.code?.coding?.[0]?.code;
+    const bCode = b.code?.coding?.[0]?.code;
+    if (!aText || !bText || !aCode || !bCode) {
       return false;
     }
-    return aDate === bDate && aText === bText;
+    return aDate === bDate && aText === bText && aCode === bCode;
```

Comment:
Shouldn't we look at all items at `coding`? In theory, two observations could have the same code, one on item 0 of the array and the other on item 1.

Reply:
Yes we should! fixed